sdk/core/azure-core-management/src/test/java/com/azure/core/management/implementation/polling/LROPollerTests.java

```diff
@@ -126,6 +126,211 @@ public void lroBasedOnProvisioningState() {
+                        getCallCount[0]++;
+                        if (getCallCount[0] < serverConfigure.pollingCountTillSuccess) {
+                            return new com.github.tomakehurst.wiremock.http.Response.Builder()
+                                .body("{\"status\": \"InProgress\"}")
+                                .build();
+                        } else if (getCallCount[0] == serverConfigure.pollingCountTillSuccess) {
+                            return new com.github.tomakehurst.wiremock.http.Response.Builder()
+                                .body("{\"status\": \"Failed\"}")
+                                .build();
+                        }
+                    } else {
+                        return new com.github.tomakehurst.wiremock.http.Response.Builder()
+                            .status(400)
+                            .body("Invalid state:" + request.getUrl())
+                            .build();
+                    }
+                }
+                return response;
+            }
+
+            @Override
+            public String getName() {
+                return "LroService";
+            }
+        };
+
+        WireMockServer lroServer = createServer(provisioningStateLroService, resourceEndpoint, operationEndpoint);
+        lroServer.start();
+
+        try {
+            final ProvisioningStateLroServiceClient client = RestProxy.create(ProvisioningStateLroServiceClient.class,
+                createHttpPipeline(lroServer.port()),
+                SERIALIZER);
+
+            PollerFlux<PollResult<FooWithProvisioningState>, FooWithProvisioningState> lroFlux
+                = PollerFactory.create(SERIALIZER,
+                new HttpPipelineBuilder().build(),
+                FooWithProvisioningState.class,
+                FooWithProvisioningState.class,
+                POLLING_DURATION,
+                newLroInitFunction(client));
+
+            int[] onNextCallCount = new int[1];
+            AsyncPollResponse<PollResult<FooWithProvisioningState>, FooWithProvisioningState> pollResponse = lroFlux.doOnNext(response -> {
+                PollResult<FooWithProvisioningState> pollResult = response.getValue();
+                Assertions.assertNotNull(pollResult);
+                onNextCallCount[0]++;
+                if (onNextCallCount[0] == 1) {
+                    Assertions.assertNotNull(pollResult.getValue());
+                    Assertions.assertEquals(LongRunningOperationStatus.IN_PROGRESS,
+                        response.getStatus());
+                } else if (onNextCallCount[0] == 2) {
+                    Assertions.assertEquals(LongRunningOperationStatus.FAILED,
+                        response.getStatus());
+                } else {
+                    throw new IllegalStateException("Poller emitted more than expected value.");
+                }
+            }).blockLast();
+
+            Assertions.assertEquals(LongRunningOperationStatus.FAILED, pollResponse.getStatus());
```

Comment:
It is only part of the complete fix for mgmt libs.

Here last `AsyncPollResponse`  will be in `FAILED` state with error in `PollResult`. Unlike track1 client-runtime, it will not throw CloudException during the polling.